inspirehep/modules/refextract/tasks.py

```diff
@@ -25,35 +25,59 @@
 from __future__ import absolute_import, division, print_function
 
 import codecs
-import contextlib
+import re
 
 from celery import shared_task
 from flask import current_app
 
 from invenio_db import db
 
 
-@contextlib.contextmanager
-def file_open_for_editing(path):
-    fd = codecs.open(path, encoding='utf-8', mode='w')
-
-    yield fd
-
-    fd.close()
+RE_ALPHANUMERIC = re.compile('\W+', re.UNICODE)
 
 
 @shared_task()
 def create_journal_kb_file():
-    result = db.session.execute("""
+    """Populate refextracts's journal KB from the DB.
+
+    Uses two raw DB queries that use syntax specific to PostgreSQL to generate
+    a file in the format that refextract expects, that is a list of lines like::
+
+        SOURCE---DESTINATION
+
+    that represents that ``SOURCE`` is translated to ``DESTINATION`` when found.
+    """
+    refextract_journal_kb_path = current_app.config['REFEXTRACT_JOURNAL_KB_PATH']
+
+    titles_query = db.session.execute("""
+        SELECT
+            r.json -> 'short_title' AS short_title,
+            r.json -> 'journal_title' -> 'title' AS journal_title
+        FROM
+            records_metadata AS r
+        WHERE
+            (r.json -> '_collections')::jsonb ? 'Journals'
+    """)
+
+    with codecs.open(refextract_journal_kb_path, encoding='utf-8', mode='w') as fd:
+        for row in titles_query:
+            fd.write(u'{}---{}\n'.format(_normalize(row['short_title']), row['short_title']))
```

Comment:
also the short title needs to be normalized.

Reply:
Uh, I thought that `DESTINATION` has to be verbatim what we want `SOURCE` to normalize to.

Reply:
you are right :+1: